### PR TITLE
Fix building issues with CUDA 10.2

### PR DESCRIPTION
**Pull request by alecyu2018**
Hello there,

Found SpMM APIs not defined for Win32 platforms in CUDA 10.2. Other than that, CUDA 10.2 comes with CUBLAS 10.2.2 which causes a version mismatch in find_cuda_config.py.

After fixed these two issues, TF is ok to build on Windows with Visual Studio 2019 on my machine.

### CODE REVIEW

**Comment by alecyu2018** [comment]
@googlebot I signed it!

**Comment by gbaned** [comment]
@alecyu2018 Thank you for your contribution. Can you please sign CLA? Thanks!

**Comment by alecyu2018** [comment]
The CLA page showed I signed it in the morning.

**Comment by alecyu2018** [comment]
@googlebot I signed it!

In case that my signed CLA is ignored for I left github username blank (since I thought I did not change it), I updated my github username in it, and so sending the "I signed it!" comment again.

**Comment by alecyu2018** [comment]
@googlebot I fixed it.

Yeah, found the e-mail in my commit not the main one.

**Comment by alecyu2018** [comment]
Hi @chsigg , any problem found?

**Comment by gbaned** [comment]
@chsigg Can you please review this PR ? Thanks!

**Comment by alecyu2018** [comment]
Another week passed. Everybody stay safe.

File: tensorflow/core/kernels/sparse/mat_mul_op.cc [hunk 1]
```diff
@@ -951,7 +951,7 @@ class CSRSparseMatrixMatVec<GPUDevice, T> {
       const int n = a.dense_shape_host(1);
       const int nnz = a.values.size();
       DCHECK_EQ(nnz, a.col_ind.size());
-#if GOOGLE_CUDA && (CUDA_VERSION >= 10020)
+#if GOOGLE_CUDA && (CUDA_VERSION >= 10020) && !defined(_WIN32)
```
Comment (chsigg) [inline tensorflow/core/kernels/sparse/mat_mul_op.cc:954]: These changes seem to undo #38802 for Windows. Is that intentional?

File: third_party/gpus/find_cuda_config.py [hunk 2]
```diff
@@ -318,10 +318,6 @@ def get_header_version(path):
     # cuBLAS uses the major version only.
     cublas_version = header_version.split(".")[0]
 
-    if not _matches_version(cuda_version, cublas_version):
```
Comment (chsigg) [inline third_party/gpus/find_cuda_config.py:321]: Is it supported to use different major version of CUDA and cuBLAS?

File: tensorflow/core/kernels/sparse/mat_mul_op.cc (hunk 1 above)
Comment (alecyu2018) [reply tensorflow/core/kernels/sparse/mat_mul_op.cc:954]: Yes, it is intentional. SpMM is not defined on Windows for CUDA 10.2.

File: third_party/gpus/find_cuda_config.py (hunk 2 above)
Comment (alecyu2018) [reply third_party/gpus/find_cuda_config.py:321]: CUDA 10.2 comes with mismatched versions between CUDA and cuBLAS.

File: .gitignore [hunk 3]
```diff
@@ -47,3 +47,6 @@ gradleBuild
 Podfile.lock
 Pods
 xcuserdata
+
+# Visual Studio
+/.vs
```
[the remark this answers is not in the document]
Comment (alecyu2018) [reply .gitignore:52]: I am ok to abandon this .gitignore change here.

File: third_party/gpus/find_cuda_config.py (hunk 2 above)
Comment (chsigg) [reply third_party/gpus/find_cuda_config.py:321]: That's not what I see for the version that we use internally. The situation might have changed.
Could you provide the exact version of the CUDA SDK that you are using? Thanks.

Comment (gbaned) [reply third_party/gpus/find_cuda_config.py:321]: @alecyu2018 Can you please check @chsigg's comments and keep us posted. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
@alecyu2018 Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**